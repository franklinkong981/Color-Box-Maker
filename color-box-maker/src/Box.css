.Box {
  margin: 10px;
}

.Box-display {
  margin: 10px auto;
}