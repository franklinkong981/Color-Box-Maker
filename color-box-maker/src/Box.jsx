import react from "react";

const Box = ({id, color, height, width, removeBox}) => {
  const handleRemove = () => {
    removeBox(id);
  };

  return (
    <div className="Box" id={id}>
      <div className="Box-display" style={{backgroundColor: color, width: width, height: height}}></div>
      <button className="Box-remove-button" onClick={handleRemove}>DELETE BOX</button>
    </div>
  );
};

export default Box;