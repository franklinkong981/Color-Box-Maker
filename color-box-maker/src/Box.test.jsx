import React from "react";
import {render} from "@testing-library/react";
import Box from "./Box.jsx";
import { expect, it } from "vitest";

const removeBox = () => {
  console.log("Box removed");
}

it("renders without crashing", function() {
  render(<Box id={1} color="blue" height={100} width={100} removBox={removeBox}/>);
});

it("matches snapshot", function() {
  const {asFragment} = render(<Box id={1} color="blue" height={100} width={100} removBox={removeBox}/>);
  expect(asFragment()).toMatchSnapshot();
});