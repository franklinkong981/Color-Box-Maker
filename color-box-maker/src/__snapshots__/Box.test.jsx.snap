// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    id="1"
  >
    <div
      class="Box-display"
      style="background-color: blue; width: 100px; height: 100px;"
    />
    <button
      class="Box-remove-button"
    >
      DELETE BOX
    </button>
  </div>
</DocumentFragment>
`;
